python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -898,3 +901,20 @@ def _get_dg_pypi_version_info(context: DgContext) -> Optional[DgPyPiVersionInfo]
                 context.config.cli.suppress_warnings,
             )
     return version_info
+
+
+def _validate_dagster_dg_and_dagster_version_compatibility(context: DgContext) -> None:
+    dagster_dg_version = Version(__version__)
+    dagster_version = context.dagster_version
+    minimum_dagster_dg_version = Version(library_version_from_core_version(str(dagster_version)))
+    if dagster_dg_version < minimum_dagster_dg_version:
+        exit_with_error(
+            f"""
+            dagster-dg version is incompatible with the installed version of dagster.
```

Comment:
I think the copy here could be wordsmithed a bit. This reads to me like a problem with the venv in which dg is running, as opposed to a compatibility problem between two different venvs.